### DISCUSSION TITLE
One file per endpoint

### DISCUSSION THREAD

**Discussion by lcfd** [vitalik/django-ninja#846] [Q&A]
Hi everyone!

I'm having some trouble with route splitting.
I'm trying to have a dedicated Python file for each endpoint but I'm not able to achieve it.
I'm ending up creating tons of routers.

Is it possible not to create routers to obtain this? 
If not what's a better approach?

Thanks!

**Comment by vitalik** [comment]
@lcfd hard to tell... maybe you can show what's you structure looks like ?

**Reply by vitalik** [reply]
and what do you call endpoint ? one operation/path - or set of operations

**Reply by lcfd** [reply]
Hi @vitalik!

By endpoint I mean a route of a router, for example: 
```python
@main_router.get("all", response=List[MySchema])
def get_list(request):
  pass
```

Since I'm not able to import an already defined router (for example `main_router = Router()`) I can't find a way to have a file per route.

What I think I want is:

`my_route_all.py`

```python

# My schemas

class MySchema(ModelSchema):
    class Config:
        model = Mine
        model_fields = "__all__"

# My route/endpoint

@main_router.get("all", response=List[MySchema])
def get_list(request):
  pass
```

I'm finding myself having a lot of dedicated schemas and a lot of routes per Django app, and navigating/reusing has become quickly uncomfortable.

Like I wrote before I don't know if it's possible and if it's a good idea in the long run.
I'm looking for hints, possibilities, and best practices.

Thank you!
